packages/core/utils/test/loadSourceMapUrl.test.js

```diff
@@ -0,0 +1,37 @@
+import assert from 'assert';
+import {matchSourceMappingURL} from '../src/loadSourceMapUrl';
+import fs from 'fs';
+import path from 'path';
+
+describe('loadSourceMap', () => {
+  it('should not match sourceMappingURL when not at the end of the bundle', () => {
+    // Code example taken from livescript.js (issue #2408 in parcel-bundler)
+    // This snippet lead to JSAsset.js being mislead and incorrectly trying to
+    // load (due to false-positive match) sourcemap before fix was introduced
+    let code = fs.readFileSync(
+      path.join(__dirname, './input/sourcemap/no-sourcemap.js'),
+      'utf-8'
+    );
+
+    assert(!matchSourceMappingURL(code));
+  });
+
+  it('should match >>referenced<< sourceMappingURL when correctly inserted at end of the bundle', () => {
```

Comment:
Very subjective option: we didn't use angle brackets to highlight words in the test titles so far